source/fundamentals/aggregation.txt

```diff
@@ -7,51 +9,235 @@ Aggregation
+- Pipeline stages have a memory limit of 100 megabytes by default. If
+  required, you may exceed this limit by using the `allowDiskUse
+  <{+api+}/mongo/options#AggregateOptions.SetAllowDiskUse>`__
+  method.
+- The :manual:`$graphLookup
+  </reference/operator/aggregation/graphLookup/>` stage
+  has a strict memory limit of 100 megabytes and ignores
+  ``allowDiskUse``.
+
+Examples
+--------
 
-     The :manual:`$graphLookup </operator/aggregation/graphLookup/>` stage has a strict memory limit of 100 megabytes
-     and will ignore ``allowDiskUse``.
+To run the examples in this section, load the sample data into the
+``tea.ratings`` collection with the following snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/aggregation.go
+   :start-after: begin insert docs
+   :end-before: end insert docs
+   :language: go
+   :dedent:
+
+Each document represents a person's rating for a particular tea, and the
+amount of visits they made to a local tea shop.
+
+Average Rating
+~~~~~~~~~~~~~~
+
+The following example calculates and displays the average rating and
+amount of ratings for each tea.
+
+The aggregation pipleline uses the ``$group`` stage to group the
+documents by the ``type`` field, calculates the average using the
+``$avg`` expression operator, and counts the number of documents using
+the ``$sum`` expression operator.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: go
+
+      // create the stage
+      groupStage := bson.D{
+          {"$group", bson.D{
+              {"_id", "$type"},
+              {"average", bson.D{
+                  {"$avg", "$rating"},
+              }},
+              {"count", bson.D{
+                  {"$sum", 1},
+              }},
+      }}}
+
+      // pass the stage into a pipeline
+      // pass the pipeline as the second paramter in the Aggregate() method
+      cursor, err := coll.Aggregate(context.TODO(), mongo.Pipeline{groupStage})
+      if err != nil {
+          panic(err)
+      }
+
+      // display the results
+      var results []bson.M
+      if err = cursor.All(context.TODO(), &results); err != nil {
+          panic(err)
+      }
+      for _, result := range results {
+          fmt.Printf("%v has an average rating of %v \n", result["_id"], result["average"])
+          fmt.Printf("%v Count: %v \n", result["_id"], result["count"])
+      }
+
+   .. output::
+      :language: go
+
+      Masala has an average rating of 8.363636363636363 
+      Masala Count: 11 
+      Earl Grey has an average rating of 7.375 
+      Earl Grey Count: 8 
+
+Omit Fields in Results
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following example matches documents with ratings over 8 for each tea
+and displays the top 5 highest ratings tea types and the amount of
+visits.
+
+The aggregation pipleline contains the following stages:
+
+- ``$match`` stage to match documents with a ``rating`` greater than 8
+- ``$unset`` stage to omit the ``_id`` and ``rating`` fields
```

Comment:
Let's do something more interesting here. What is visits supposed to represent? Are these documents per user?